manifests/repo.pp

```diff
@@ -143,11 +143,11 @@
           require => Exec["import_gpg_${normalized_name}"],
         }
 
-        exec { "yum_make_cache_${repo_name}":
-          command => "yum -q makecache -y --disablerepo='*' --enablerepo='${normalized_name}'",
-          path    => '/usr/bin',
-          require => File[$normalized_name],
```

Comment:
A better solution would be to change this to a refresh_only, and changing the require to a subscribe, then the exec only runs if the repo is changed.